model/implementation/DeliveryService.php

```diff
@@ -90,7 +92,32 @@ public function getProctorableDeliveries(User $proctor, $options = array())
         }
         return $allDeliveries;
     }
-    
+
+    /**
+     * Gets all deliveries available for a test center
+     * @param string|core_kernel_classes_Resource $testCenterId
+     * @return \taoDelivery_models_classes_DeliveryRdf[]
+     */
+    public function getTestCenterDeliveries($testCenterId)
+    {
+        if (is_object($testCenterId) && $testCenterId instanceof \core_kernel_classes_Resource) {
+            $testCenter = $testCenterId;
+            $testCenterId = $testCenter->getUri();
+        }
+
+        if (is_string($testCenterId)) {
+            $testCenter = new \core_kernel_classes_Resource($testCenterId);
+        }
```

Comment:
strange construction for the resource $testCenter : what about only contructing the Resource if the testCenterId is a string ? If it is already a Resource, you can simply assign it to the variable $testCenter